src/snowflake/snowpark/_internal/open_telemetry_prototype.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+import inspect
+from typing import Dict
+
+# this parameter make sure no error when open telemetry is not installed
+open_telemetry_not_found = False
+try:
+    from opentelemetry import trace
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+except ImportError:
+    open_telemetry_not_found = True
+
+if not open_telemetry_not_found:
+    resource = Resource(attributes={SERVICE_NAME: "snowpark-python-open-telemetry"})
+    tracer = trace.get_tracer("snowflake.snowpark.dataframe")
+
+
+def open_telemetry(name):
```

Comment:
This decorator needs type annotations. Please ask @sfc-gh-stan  how to type-annotate a decorator.